### DISCUSSION TITLE
Ender 3 S1 - F1 with CR touch.

### DISCUSSION THREAD

**Discussion by dbeistel** [mriscoc/Ender3V2S1#658] [General]
I recently ( Sunday ) came to own an Ender 3 S1. It is running stock Firmware. I was going to update it, but if figured I would look for things like that have been done with better thought behind them.  Like the marlin.crc.id.au site, I ran across your setup and liked what I saw. I am looking to load it on the E3S1f1 I have with a crTouch and based on your description of what to download I should be looking for something named Ender3S1-F1-(Date).bin all I see in your list for the E3S1 is for the F4 with that naming structure. Is there something I am missing? Below are the listed files for the S1 I can find. Thank you in advance for help. 

[Ender3S1-F1-UBL-MPC-20221222.bin]
[Ender3S1-F4-20221222.bin]
[Ender3S1-F4-UBL-20221222.bin]

**Comment by mriscoc** [comment]
https://github.com/mriscoc/Ender3V2S1/wiki#versions

**Reply by dbeistel** [reply]
Yep, I read that part about five times. 

On that page for the S1 you have three versions listed; none match the naming for what you have listed in the version section.  The only one that is an F1 is the ULB.  I do not have ULB I am CR Touch. 
![Screenshot 2023-01-20 125703](https://user-images.githubusercontent.com/20329051/213794383-198eb2fe-aa64-445e-8fd2-57753845e668.png)

**Reply by TheWebMachine** [reply]
UBL only refers to the type of bed leveling in the firmware. Instead of the "legacy" ABL/LA type of leveling code/algorithms, it is using Marlin's newer Universal Bed Leveling code.

By design, this implies you have a BL or CR Touch, or some other probe installed.

Edit: I do agree the page could use a little better clarification as the pre-built firmwares list has evolved a bit.

**Reply by mriscoc** [reply]
From the link: https://github.com/mriscoc/Ender3V2S1/wiki#versions

![image](https://user-images.githubusercontent.com/2745567/213801768-5ded9e43-22ea-44bf-aec8-bfb33b36124b.png)

![image](https://user-images.githubusercontent.com/2745567/213801657-566a800e-332b-4391-aa61-b8ad0b145f0f.png)

![image](https://user-images.githubusercontent.com/2745567/213801701-fa0a39e9-64b0-4176-9097-33da84225720.png)

**Reply by TheWebMachine** [reply]
Maybe mention in the UBL paragraph the probe support so UBL doesn't *seem* different from the E3S1/CR Touch mention to those who aren't as knowledgeable? I know I originally had to read through it a couple times to wrap my head around which firmware I needed/wanted, and I'm far from a novice.

Perhaps a table-style feature matix of the current releases for the more visually-inclined.

**Reply by TheWebMachine** [reply]
Actually, a table feature matrix would be awesome now that I think about it. I had to dig through multiple discussions to find out things like how the F4 version had to have things like BFT removed to make room for UBL. A matrix table showing what is added or removed from each release would be very helpful. Most of it is buried within discussion responses and comments right now, making it very hard to find.

**Reply by dbeistel** [reply]
Okay thank you, That helps me understand.  I have gotten burned a few times on E3V2 firmware. because I was not liberal with my definition of what the information said.  I have had to replace my MB once because I bricked it loading a version that was not correct.  My interpretation about UBL was the pressure Bed leveling method like the CR-10 Smart (which it is not so), not BR/BL touch.

**Comment by dbeistel** [comment]
I am not sure if could use your FW on E3V2 because the BL touch port on my MB doesn't work correctly and my BLT is plugged into the Z-Axis port.  I had to build that myself and I don't want to do that again.

**Reply by mriscoc** [reply]
The precompiled firmware for CR/3D/BL Touch only works with the probe port: https://github.com/mriscoc/Ender3V2S1/wiki/3D-BLTouch
